src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -429,19 +428,27 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
                 diagnostics,
                 useMethodSignatureParameterTypes: !(methodSymbol is null),
                 methodSignatureOpt: methodSymbol,
-                returnTypes: null,
                 initialState: null,
-                analyzedNullabilityMapOpt: analyzedNullabilities);
+                analyzedNullabilityMapOpt: analyzedNullabilities,
+                takeIncrementalSnapshots: createSnapshots,
+                ref snapshotManager,
+                ref ignoredReturnTypes);
 
             var analyzedNullabilitiesMap = analyzedNullabilities.ToImmutableDictionaryAndFree();
 
+            snapshots = createSnapshots ? snapshotManager.Snapshots : null;
+            Debug.Assert((!createSnapshots && snapshots == null)
+                         || snapshots != null);
+
 #if DEBUG
             // https://github.com/dotnet/roslyn/issues/34993 Enable for all calls
             if (compilation.NullableAnalysisEnabled)
             {
-                DebugVerifier.Verify(analyzedNullabilitiesMap, node);
+                DebugVerifier.Verify(analyzedNullabilitiesMap, snapshots, node);
             }
 #endif
+
+            snapshotManager.Free();
```

Comment:
>snapshotManager [](start = 12, length = 15)

Is this never null?

Reply:
No longer relevant.

---
In reply to: [288785768](https://github.com/dotnet/roslyn/pull/35850#discussion_r288785768) [](ancestors = 288785768)